jetstream/resource_jetstream_stream.go

```diff
@@ -403,39 +413,18 @@ func resourceStreamRead(d *schema.ResourceData, m any) error {
 
 	if str.IsMirror() {
 		mirror := str.Mirror()
-		d.Set("mirror.0.name", mirror.Name)
-		d.Set("mirror.0.filter_subject", mirror.FilterSubject)
-		d.Set("mirror.0.start_seq", mirror.OptStartSeq)
-		if mirror.OptStartTime != nil {
-			d.Set("mirror.0.start_time", mirror.OptStartTime.Format(time.RFC3339))
-		}
-		if mirror.External != nil {
-			d.Set("mirror.0.external.api", mirror.External.ApiPrefix)
-			d.Set("mirror.0.external.deliver", mirror.External.DeliverPrefix)
-		}
-		for c, v := range mirror.SubjectTransforms {
-			d.Set(fmt.Sprintf("mirror.0.subject_transforms.%d.src", c), v.Source)
-			d.Set(fmt.Sprintf("mirror.0.subject_transforms.%d.dest", c), v.Destination)
+		mirrors := []map[string]any{
+			streamSourceConfigRead(mirror),
 		}
+		d.Set("mirror", mirrors)
```

Comment:
should maybe guard against len(mirrors)!=1

Reply:
Again leaving the comment for posterity, since we're setting `mirrors` from a valid streamconfig and the resource validates that the list is only 1 element, we don't need to check the size again.